### PR TITLE
Liking of your own stuff wasn't possible/enable dislike for Diaspora

### PR DESCRIPTION
**Pull request by annando**
Diaspora doesn't display the dislike, but does transport it.

### CODE REVIEW

File: src/Object/Post.php [hunk 1]
```diff
@@ -82,8 +82,9 @@ public function __construct(array $data)
 					continue;
 				}
 
-				// You can always comment on Diaspora items
-				if (($item['network'] == NETWORK_DIASPORA) && (local_user() == $item['uid'])) {
+				// You can always comment on Diaspora and OStatus items
+				// The empty network can happen with your local (self) contact
+				if (in_array($item['network'], ['', NETWORK_OSTATUS, NETWORK_DIASPORA]) && (local_user() == $item['uid'])) {
```
Comment (MrPetovan) [inline src/Object/Post.php]: Is there any way to ensure the `'network'` value is among the available ones?

Comment (annando) [reply src/Object/Post.php]: I don't understand?

Comment (MrPetovan) [reply src/Object/Post.php]: `$item['nework']` should always be amongst the expected values, can we ensure this before this check?

Comment (annando) [reply src/Object/Post.php]: This is more or less a bugfixing. Normally the "writable" value is defined by the contact. On Twitter and other external networks this value was always set to "true" (so no need to change here). But for Diaspora this wasn't the case - although technically you could always comment. So we just fix it here.

And with the empty network we just correct the problem that the "self" contact hasn't set this value as well.

This is some temporarily solution and I guess we will eventually find a better way.

Comment (annando) [reply src/Object/Post.php]: To be clear: We only **overwrite** the contact value.

Comment (MrPetovan) [reply src/Object/Post.php]: Ok, as long as you keep it in mind.

Comment (annando) [reply src/Object/Post.php]: You can be sure of this. Just think of "last-child". This is something similar that I have to address during the item structure reworking.

**Pull request merged**